cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/dc/DCIslandEnd.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+package com.powsybl.cgmes.conversion.elements.dc;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author Romain Courtier {@literal <romain.courtier at rte-france.com>}
+ */
+public record DCIslandEnd(Set<DCEquipment> dcEquipments) {
+
+    boolean isAdjacentTo(DCIslandEnd otherDcIslandEnd) {
+        // Two DCIslandEnd are adjacent if they share a DCLineSegment.
+        Set<DCEquipment> commonDcLineSegments = dcEquipments.stream()
+                .filter(DCEquipment::isLine)
+                .collect(Collectors.toSet());
+        commonDcLineSegments.retainAll(otherDcIslandEnd.dcEquipments);
+        return !commonDcLineSegments.isEmpty();
+    }
+
+    public List<DCEquipment> getAcDcConverters() {
+        return dcEquipments.stream()
+                .filter(DCEquipment::isConverter)
+                .sorted(Comparator.comparing(DCEquipment::id))
+                .toList();
+    }
+
+    public List<DCEquipment> getDcLineSegments() {
+        return dcEquipments.stream()
```

Comment:
Please, could you add a comment explaining why the sorting is done in descending order?